### ISSUE TITLE
Conversion to first-order ODEs

### ISSUE THREAD

**Issue by ChrisRackauckas** [JuliaDiffEq/SciCompDSL.jl#17]
This DSL has enough information that we could add detection of higher order ODEs and then make it convert down to a system of first order ODEs for the ODE solvers.

**Comment by ChrisRackauckas** [comment]
It should have a default argument `naming_scheme = "_"`, so that `u''' => u_ttt` or something like that. With a choice of naming scheme the implementation should be pretty straight forward. It should make sure to substitute out any higher order derivative as well. So a good test equation would be:

```julia
u''' = 2u'' + u' + x' + 1
x'' = x' + 2
```

which should give the system:

```julia
@Derivs D'~t
#u_tt = u''
#u_t = u'
[D*u_tt = 2u_tt + u_t + x_t + 1,
D*u_t = u_tt,
D*u = u_t,
D*x_t = x_t + 2,
D*x = x_t]
```

**Comment by YingboMa** [comment]
Closed by #40
